modules/opengl/src/image/imagecompositor.cpp

```diff
@@ -53,14 +64,19 @@ void ImageCompositor::composite(const Image& source, Image& target, ImageType ty
 
 void ImageCompositor::composite(ImageInport& source, ImageOutport& target, ImageType type) {
     if (source.isReady() && target.hasData()) {
+        copyTextures(*target.getData());
```

Comment:
is it not possible to call composite(*source.getData(), *target.getData(), type) to avoid code duplication?

Reply:
not really. `target.getData()` returns a `std::shared_ptr<const Image>`, note the `const` here. Hence, it will not be possible to render into the image.